components/image/image.component.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
+ */
+
+import { DOCUMENT, isPlatformServer } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  Inject,
+  Input,
+  OnChanges,
+  OnDestroy,
+  OnInit,
+  PLATFORM_ID,
+  Renderer2,
+  SimpleChanges,
+  ViewEncapsulation
+} from '@angular/core';
+import { NzConfigKey, NzConfigService, WithConfig } from 'ng-zorro-antd/core/config';
+import { BooleanInput, NzSafeAny } from 'ng-zorro-antd/core/types';
+import { InputBoolean } from 'ng-zorro-antd/core/util';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { defaultLoader, NzImageLoader } from './image-loader';
+
+const NZ_CONFIG_MODULE_NAME: NzConfigKey = 'image';
+
+@Component({
+  selector: 'nz-image',
+  exportAs: 'nzImg',
+  template: `
+    <img [attr.src]="src" [attr.width]="width" [attr.height]="height" [attr.srcset]="srcSet" [attr.sizes]="sizes" [attr.alt]="nzAlt" />
+  `,
+  preserveWhitespaces: false,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None
+})
+export class NzImageComponent implements OnInit, OnChanges, OnDestroy {
+  readonly _nzModuleName: NzConfigKey = NZ_CONFIG_MODULE_NAME;
+  static ngAcceptInputType_nzOptimize: BooleanInput;
+  static ngAcceptInputType_nzPreload: BooleanInput;
+
+  @Input() nzSrc: string = '';
+  @Input() nzAlt: string = '';
+  @Input() nzWidth: string | number = 'auto';
+  @Input() nzHeight: string | number = 'auto';
+  @Input() @WithConfig() nzLoader: NzImageLoader = defaultLoader;
+  @Input() @InputBoolean() @WithConfig() nzOptimize: boolean = false;
+  @Input() @InputBoolean() @WithConfig() nzPreload: boolean = false;
+  @Input() @WithConfig() nzOptimizeSizes: number[] = [16, 32, 48, 64, 96, 128, 256, 384, 640, 750, 828, 1080, 1200, 1920, 2048, 3840];
+
+  get width(): number {
+    return typeof this.nzWidth === 'number' ? this.nzWidth : parseInt(this.nzWidth, 10);
```

Comment:
The `nzWidth` default value is `auto` will be return `NaN` (`parseInt('auto', 10)`).